File: 1749. Maximum Absolute Sum of Any Subarray/main.go
package main

import "math"

func maxAbsoluteSum(nums []int) int {
	min_pre, max_pre, cur, res := 0, 0, 0, 0
	for _, n := range nums {
		cur += n
		res = max(res, int(math.Abs(float64(cur-min_pre))), int(math.Abs(float64(cur-max_pre))))
		min_pre = min(min_pre, cur)
		max_pre = max(max_pre, cur)
	}
	return res
}

File: 1749. Maximum Absolute Sum of Any Subarray/main.py
# total sam

from typing import List


class Solution:
    def maxAbsoluteSum(self, nums: List[int]) -> int:
        max_positive, min_negative = nums[0], nums[0]
        prefix = [[n, n] for n in nums]
        for i in range(1, len(nums)):
            prefix[i][0] = max(prefix[i][0], prefix[i - 1][0] + nums[i])
            max_positive = max(max_positive, prefix[i][0])
            prefix[i][1] = min(prefix[i][1], prefix[i - 1][1] + nums[i])
            min_negative = min(min_negative, prefix[i][1])

        return max(max_positive, -min_negative)